File: packages/python-client/examples/01_quickstart.py
"""Smallest possible governed call.

    pip install agentvalet
    agentvalet register --code <invite-or-enrollment-code>
    python 01_quickstart.py

The point: there is no Slack token in this file, in your environment, or in
your process memory. AgentValet holds it, checks the call against what your
owner granted, injects the credential at request time, and writes an audit
record. Your agent only ever proves who it is.
"""

from agentvalet import AgentValet, AccessDeniedError

# from_env() needs no arguments on a machine that has run `agentvalet register`.
# It reads AGENTVALET_AGENT_ID / AGENTVALET_OWNER_ID (or the bare AGENT_ID /
# OWNER_ID) and finds the key at ~/.agentvalet/agent.key.
with AgentValet.from_env() as av:

    # What is this agent actually allowed to do? Deny-by-default: anything not
    # listed here will raise AccessDeniedError.
    grants = av.list_platforms()
    print("granted:", grants)

    try:
        result = av.call(
            platform="slack",
            endpoint="/api/chat.postMessage",
            method="POST",
            scope="chat:write",
            data={"channel": "#general", "text": "Deploy finished."},
            # Shown to whoever approves, if this scope is approval-gated.
            reason="Notify the team that the deploy completed",
        )
        print("sent:", result)

    except AccessDeniedError as err:
        # Not a crash — a governance decision your agent can act on.
        print(f"no grant for {err.platform}:{err.scope}")
        print("asking an admin...")
        decision = av.request_access(
            platform="slack",
            scope="chat:write",
            reason="Post deploy notifications to #general",
        )
        print("decision:", decision["status"])

File: packages/client/examples/01-quickstart.ts
/**
 * Smallest possible governed call.
 *
 *   npm install @agentvalet/client
 *   npx @agentvalet/register
 *   npx tsx 01-quickstart.ts
 *
 * The point: there is no Slack token in this file, in your environment, or in
 * your process memory. AgentValet holds it, checks the call against what your
 * owner granted, injects the credential at request time, and writes an audit
 * record. Your agent only ever proves who it is.
 *
 * If you're inside Claude Code, Claude Desktop or Cursor, you don't need this
 * package at all — install @agentvalet/mcp-server and call use_platform.
 */

import { AgentValet, AccessDeniedError } from "@agentvalet/client";

// fromEnv() needs no arguments on a machine that has run `npx @agentvalet/register`.
const av = AgentValet.fromEnv();

// What is this agent actually allowed to do? Deny-by-default: anything not
// listed here throws AccessDeniedError.
console.log("granted:", await av.listPlatforms());

try {
  const result = await av.call({
    platform: "slack",
    endpoint: "/api/chat.postMessage",
    method: "POST",
    scope: "chat:write",
    data: { channel: "#general", text: "Deploy finished." },
    // Shown to whoever approves, if this scope is approval-gated.
    reason: "Notify the team that the deploy completed",
  });
  console.log("sent:", result);
} catch (err) {
  if (err instanceof AccessDeniedError) {
    // Not a crash — a governance decision your agent can act on.
    console.log(`no grant for ${err.platform}:${err.scope}, asking an admin...`);
    const decision = await av.requestAccess({
      platform: "slack",
      scope: "chat:write",
      reason: "Post deploy notifications to #general",
    });
    console.log("decision:", decision.status);
  } else {
    throw err;
  }
}
